docs/mmds/mmds-user-guide.md

```diff
@@ -0,0 +1,240 @@
+                       "local-hostname": "ip-10-251-50-12.ec2.internal",
+                       "public-hostname": "ec2-203-0-113-25.compute-1.amazonaws.com",
+                       "network": {
+                            "interfaces": {
+                                 "macs": {
+                                      "02:29:96:8f:6a:2d": {
+                                           "device-number": "13345342",
+                                           "local-hostname": "localhost",
+                                           "subnet-id": "subnet-be9b61d"
+                                      }
+                                 }
+                            }
+                       }
+                  }
+            }      
+    }"
+```
+
+To partially update existing metadata, a HTTP `PATCH` request to the `/mmds`
+resource has to be issued, using as a payload the metadata patch, as
+[JSON Merge Patch]((https://tools.ietf.org/html/rfc7396)) functionality
+describes. A complete description of updating metadata firecracker API can be
+found in the [firecracker swagger file](../../src/api_server/swagger/firecracker.yaml).
+
+### Example
+
+```bash
+curl --unix-socket /tmp/firecracker.socket -i \
+    -X PATCH "http://localhost/mmds" \
+    -H "accept: application/json" \
+    -H "Content-Type: application/json" \
+    -d "{
+            "latest": {
+                  "meta-data": {
+                       "ami-id": "ami-87654321",
+                       "reservation-id": "r-79054aef",
+                  }
+            }      
+    }"
+```
+
+# Retrieving metadata
+
+MicroVM metadata can be retrieved:
+* from host operating system, communicating with the firecracker API server.
+* from guest operating system, communicating directly with MMDS.
+
+For the scope of this chapter, let's assume the data store content is the JSON
+below:
+
+```json
+{
+    "latest": {
+          "meta-data": {
+               "ami-id": "ami-87654321",
+               "reservation-id": "r-79054aef"
+          }
+    }
+}  
+```
+
+## Retrieving metadata in the host operating system
+
+To retrieve existing metadata in host operating system, a HTTP `GET` request to
+the `/mmds` resource must be issued. The HTTP response returns the existing
+metadata, structured in JSON format. A complete description of
+retrieving metadata firecracker API can be found in the
+[firecracker swagger file](../../src/api_server/swagger/firecracker.yaml).
+
+### Example
+
+```bash
+curl -s --unix-socket /tmp/firecracker.socket http://localhost/mmds
+```
+
+Output:
+
+```json
+{
+    "latest": {
+          "meta-data": {
+               "ami-id": "ami-87654321",
+               "reservation-id": "r-79054aef"
+          }
+    }
+}  
+```
+
+## Retrieving metadata in the guest operating system
+
+To retrieve existing metadata in guest operating system, a HTTP `GET` request
+must be issued. The requested resource can be referenced by its corresponding
```

Comment:
A HTTP get must be issued to the configured MMDS IP. Worth to add more examples, like getting a leaf, or getting from a list - like Fargate query for container ID where they were getting just a container ID

Reply:
We can add few other examples to let users know how they can make use of `JSON Pointer` syntax. However, we should make sure users are encouraged to rely on the proper documentation from which they can learn how to reference JSON values using JSON pointers. I think the standard does a better job in this matter, and that is why I would suggest to keep only one simple example in this documentation, to show that the JSON pointer is the actual `GET` request path, and not stress out how to use/form the JSON pointers to reference different values from an example data store. WDYT?

Reply:
I think is important to cover with examples on how MMDS reponds both to get on objects and on leafs